src/agent.rs

```diff
@@ -483,20 +496,31 @@ pub struct PublicBridge<T: Agent> {
     worker: Value,
     id: HandlerId,
     _agent: PhantomData<T>,
+
 }
 
 impl<AGN: Agent> PublicBridge<AGN> {
     fn send_to_remote(&self, msg: ToWorker<AGN::Input>) {
-        // TODO Important! Implement.
-        // Use a queue to collect a messages if an instance is not ready
-        // and send them to an agent when it will reported readiness.
         let msg = msg.pack();
-        let worker = &self.worker;
-        js! {
-            var worker = @{worker};
-            var bytes = @{msg};
-            worker.postMessage(bytes);
-        };
+        if self.worker_is_loaded() {
+            let worker = &self.worker;
+            js! {
+                var worker = @{worker};
+                var bytes = @{msg};
+                worker.postMessage(bytes);
+            };
+        } else {
+            self.msg_to_queue(msg);
+        }
+    }
+    fn worker_is_loaded(&self) -> bool { DEDICATED_WORKERS_LOADED.with(|local| local.borrow().contains(AGN::name_of_resource())) }
+    fn msg_to_queue(&self, msg: Vec<u8>) {
+        DEDICATED_WORKERS_EARLY_MSGS_QUEUE.with(|local|
+            match local.borrow_mut().entry(AGN::name_of_resource()) {
+                std::collections::hash_map::Entry::Vacant(record) => {record.insert({let mut v = Vec::new(); v.push(msg); v});},
```

Comment:
You can do `use std::collections::hash_map::Entry as HashMapEntry` to remove the long path